### ISSUE TITLE
Failed to upgrade to @remirror/core v0.8.0

### ISSUE THREAD

**Issue by zhaoyao91** [remirror/remirror#237]
## Description

I've upgraded remirror from the previous version to the latest one, and fixed a type from `CommondFunction` to `ProsemirrorCommandFunction`, but my extision just cannot work with strange error:

```
variable-placeholder-extension.ts:12 Uncaught TypeError: Class constructor NodeExtension cannot be invoked without 'new'
    at new VariablePlaceholderExtension (variable-placeholder-extension.ts:12)
    at remirror-manager.tsx:107
    at Array.forEach (<anonymous>)
    at RemirrorManager.get manager [as manager] (remirror-manager.tsx:101)
    at RemirrorManager.render (remirror-manager.tsx:128)
    at finishClassComponent (react-dom.development.js:18470)
    at updateClassComponent (react-dom.development.js:18423)
    at beginWork$1 (react-dom.development.js:20186)
    at HTMLUnknownElement.callCallback (react-dom.development.js:336)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:385)
```

My update is 

- @remirror/core: 0.7.4 => 0.8.0
- @remirror/core-extensions: 0.7.4 => 0.7.5
- @remirror/react: 0.7.5 => 0.7.6

Here is the full code of the extension:

```
import {
  Attrs,
  CommandNodeTypeParams,
  NodeExtension,
  NodeExtensionSpec,
  ProsemirrorCommandFunction
} from "@remirror/core";

export const TAG_NAME = "span" as const;
export const NODE_TYPE = "variable-placeholder" as const;

export class VariablePlaceholderExtension extends NodeExtension {
  get name() {
    return "variablePlaceholder" as const;
  }

  get schema(): NodeExtensionSpec {
    return {
      attrs: {
        ...this.extraAttrs(),
        key: { default: "variable" },
        text: { default: "placeholder" }
      },

      inline: true,
      group: "inline",
      atom: true,

      toDOM: node => [
        TAG_NAME,
        {
          "data-node-type": NODE_TYPE,
          "data-attr-key": node.attrs.key,
          style:
            "display: inline-block; border-radius: 2px; padding: 0 8px; margin: 1px; background-color: #e1eaff; color: #133c9a; line-height: calc(1.5em - 2px);"
        },
        node.attrs.text
      ],

      parseDOM: [
        {
          tag: `${TAG_NAME}[data-node-type="${NODE_TYPE}"]`,
          getAttrs: (dom: any) => {
            return {
              key: dom.getAttribute("data-attr-key"),
              text: dom.textContent
            };
          }
        }
      ]
    };
  }

  public commands({ type }: CommandNodeTypeParams) {
    return {
      insertVariablePlaceholder: (
        attrs?: Attrs
      ): ProsemirrorCommandFunction => (state, dispatch) => {
        const node = type.create(attrs);
        const transaction = state.tr.replaceSelectionWith(node, true);

        if (dispatch) {
          dispatch(transaction);
        }

        return true;
      }
    };
  }
}

```

Here are how the extension being used: 

```
export const EmailContentEditor: FC<Props> = ({
  initialContent,
  onContentChange,
  onHTMLChange,
  variables = []
}) => {
  const events = useMemo(() => new EventEmitter(), []);

  const handleLinkActivation = useCallback(() => {
    events.emit("activateLink");
  }, [events]);

  const handleChange = useChangeHandler({
    initialContent,
    onContentChange,
    onHTMLChange
  });

  return (
    <RemirrorManager>
      <RemirrorExtension Constructor={BoldExtension} />
      <RemirrorExtension Constructor={ItalicExtension} />
      <RemirrorExtension Constructor={UnderlineExtension} />
      <RemirrorExtension Constructor={StrikeExtension} />
      <RemirrorExtension Constructor={HeadingExtension} levels={[1, 2, 3]} />
      <RemirrorExtension Constructor={ListItemExtension} />
      <RemirrorExtension Constructor={BulletListExtension} />
      <RemirrorExtension Constructor={OrderedListExtension} />
      <RemirrorExtension
        Constructor={LinkExtension}
        activationHandler={handleLinkActivation}
      />
      <RemirrorExtension Constructor={VariablePlaceholderExtension} />
      <ManagedRemirrorProvider
        initialContent={initialContent}
        onChange={handleChange}
      >
        <Fragment>
          <EditorLayout
            menu={<EditorMenu events={events} variables={variables} />}
            editor={<InnerEditor />}
          />
          <EventsHandler events={events} />
        </Fragment>
      </ManagedRemirrorProvider>
    </RemirrorManager>
  );
};

```

after remove my extension, it works, but by checking the source of other built-in extension, I can't find out the reason of the broken.

## Checklist

- [ ] I have read the [**contributing**](https://github.com/remirror/remirror/blob/canary/docs/contributing.md) document.

**Comment by zhaoyao91** [comment]
even stranger, I just copy the `BulletListExtension` and rename it as `EmptyExtension`, and use it to replace the original BulletListExtension, it won't work neither!

**Comment by ifiokjr** [comment]
@zhaoyao91 I've verified the bug and it's an annoying one. 

It seems that our `browserlist` configuration is changing the way the code is rendering and outputting classes rather than functions.

https://github.com/remirror/remirror/blob/2a79b20fa7406a802273e7c61e35fe5d26fdf0f2/package.json#L171-L178

The extension from the [`@remirror/core@0.8.0`](https://unpkg.com/browse/@remirror/core@0.8.0/lib/node-extension.js) release is using the `class` keyword while the previous version [`@remirror/core@0.7.4`](https://unpkg.com/browse/@remirror/core@0.7.4/lib/node-extension.js) is not.

I assume your build process is using a default build process and hence transforming all classes into functions. This is causing the error that you're seeing.

Try adding the exact same `browserlist` configuration to your `package.json` to see if it resolves the immediate issue.

I'm stuck between two minds about changing this since I've never planned to support es5 only browsers but it might be hard for users to integrate with current apps if they need to add extra configuration with browserlist.

**Comment by ifiokjr** [comment]
It's actually an interesting dilemma. The current JavaScript ecosystem embraces the latest code and newest language features via tools like babel, but when actually trying to output some of these features in the browser, most frameworks like `NextJS` (and any framework using the default browser settings of `@babel-preset-env`) default to producing `ES5` code which isn't compatible with some `ES6` features like extending classes.

**Comment by zhaoyao91** [comment]
adding  browserslist works for apps.

I am building a library using tsdx, it will generate es and cjs module while target es5, but it won't transpile the dependencies and the final app will have to recognize and bundle them.

it's ok for me to add some browserslist in the apps since these apps are just playground, but my library will pass this problem to the final user which create app using our libs. we have to **ALERT** them in some place to add the browserslist and exclude the users who maintained some old app and the toolchains are not easy to change.

you are right, the change may push forward the ecosystem, but hurt some users. I have no idea which  way is better.

**Comment by zhaoyao91** [comment]
More problems found in this change. 

- I have to update the lib's browserslist to generate codes which can consume your code, and this requirements is in fact propogated to the final user, force them to update their app's settings.
- The browserslist you provided is actually too narrow, if we have to keep the pace, the defaults may be a proper option.
- Now my playground app can only comsume the generated code, not the src code of my lib, but it can do that before.

I have some prefecence this time. We are library authors, cathing the fashion things may be an option for end users / app creators, but for us the compatibility and usability is more important. We indeed need to keep the pace of the eco, but perhaps with a safer one.

**Comment by ifiokjr** [comment]
Makes sense. I'll make a patch fix later on this evening.

**Comment by tommoor** [comment]
I believe this is broken again on the `@next` version, I get the same error attempting to create a custom extension that extends NodeExtension

**Comment by ocavue** [comment]
Hi @tommoor . Can you try to add `"browserslist": ["since 2017"]` in your `package.json` and see if there are still problems?

Quota from the [document](https://remirror.io/docs/guide/installation#browser-support): 

> The main reason for this configuration is to support class syntax properly. Projects like `next.js` automatically compile your code down to `es5` which causes problems when extending classes. If you don't plan on creating your own extensions or presets, you can ignore this requirement.

**Comment by tommoor** [comment]
No luck with that unfortunately

**Comment by tommoor** [comment]
Upgrading babel fixed the issue, looks like there is a bug that was fixed in `7.10`
https://github.com/babel/babel/commit/f6c7bf36cec81baaba8c37e572985bb59ca334b1
